sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +246,97 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
+   */
+  manualReassessment(applicationId: number, note: string, userId: number) {
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const currentUser = { id: userId } as User;
+      const now = new Date();
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      const application = await applicationRepo.findOne({
+        select: {
+          id: true,
+          currentAssessment: true as unknown,
+          student: { id: true },
+        },
+        relations: [
+          "currentAssessment",
+          "currentAssessment.offering",
+          "currentAssessment.studentAppeal",
+          "student",
+        ],
+        where: {
+          id: applicationId,
+        },
+      });
+
+      if (!application) {
+        throw new CustomNamedError(
+          `Application not found.`,
+          APPLICATION_NOT_FOUND,
+        );
+      }
+      if (
+        application.currentAssessment.studentAssessmentStatus !==
+        StudentAssessmentStatus.Completed
+      ) {
+        throw new CustomNamedError(
+          `Application current assessment expected to be '${StudentAssessmentStatus.Completed}' to allow manual reassessment.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+      if (application.isArchived) {
+        throw new CustomNamedError(
+          `Application cannot have manual reassessment after being archived.`,
+          INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+        );
+      }
+
+      // Adds application note.
+      const newNote = new Note();
+      newNote.description = note;
+      newNote.noteType = NoteType.Application;
+      newNote.creator = currentUser;
+      newNote.createdAt = now;
+      const savedNote = await transactionalEntityManager
+        .getRepository(Note)
+        .save(newNote);
+      // Associates the created note with the student.
+      await transactionalEntityManager
+        .getRepository(Student)
+        .createQueryBuilder()
+        .relation(Student, "notes")
+        .of(application.student)
+        .add(savedNote);
+
+      const oldCurrentAssessment = application.currentAssessment;
+      const newCurrentAssessment = new StudentAssessment();
+      newCurrentAssessment.triggerType =
+        AssessmentTriggerType.ManualReassessment;
+      newCurrentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.Submitted;
+      newCurrentAssessment.studentAssessmentStatusUpdatedOn = now;
+      newCurrentAssessment.offering = oldCurrentAssessment.offering;
+      newCurrentAssessment.studentAppeal = oldCurrentAssessment.studentAppeal;
+      newCurrentAssessment.submittedDate = now;
+      newCurrentAssessment.submittedBy = currentUser;
+      newCurrentAssessment.createdAt = now;
+      newCurrentAssessment.creator = currentUser;
+      newCurrentAssessment.application = application;
+
+      await transactionalEntityManager
```

Comment:
Can we save this in a similar way as the other reassessments? Only one call to `save` is happening.
https://github.com/bcgov/SIMS/blob/e8dc452c773c5a01c63c2072f98f668e328a6568/sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts#L97-L117

Student Appeal also is similar.
https://github.com/bcgov/SIMS/blob/ff9c49b2e68edfa76784a644059e79c841f3bcd1/sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts#L406-L426

Reply:
I agree. Save will cascade save the application and assessment both.